Hello_Object_Oriented_Due_Dec_1/renderBoard.js

```diff
@@ -0,0 +1,132 @@
+
+
+class Task {
+  constructor(taskName='', assignee='') {
+    this.taskName = taskName;
+    this.assignee = assignee;
+  }
+
+  render() {
+    if (this.assignee != '' && this.taskName != '') {
+
+      return `${this.taskName} ${this.assignee}`;
+    }
+    else if (this.taskName != '') {
+
+      return `${this.taskName}`;
+    } else {
+      return '';
+    }
+  }
+}
+
+
+class List {
+
+  constructor(listName) {
+    this.listName = listName;
+    this.lists = []; // store tasks
+  }
+  addTask (task) {
+    this.lists.push(task);
+    return this;
+  }
+  removeTask(task) {
+    for (var i = 0; i < this.lists.length; i++) {
+      if (this.lists[i].listName === task) {
+        // console.log(`>>>> ${this.taskList[i].taskName}`);
+        return this.lists.splice(i,1).pop();
+        // return `The task "${task}" was removed!`;
+      } else {
+        return null;
+      }
+    }
+  }
+
+  render(){
+    let dash = '|---------------------';
+    let result = '';
+    result += `${dash}\n|${this.listName}\n${dash}`;
+    // console.log(this.lists[2].listName);
+    for(let i in this.lists) {
+      if (this.lists[i].assignee !== '') {
```

Comment:
`this.lists` (which should be called `this.tasks`) is a list of `Task` instances.
Every instance of `Task` has its own `render` method. It was defined in the `Task` class.
You should be using this render method to render the task
```javascript
/*
if (this.lists[i].assignee !== '') {
   result += `\n|${i}> ${this.lists[i].listName} • ${this.lists[i].assignee}`;
} else {
   result += `\n|${i}> ${this.lists[i].listName}`;
}

^ all of this code can be replaced with the one line below
Since the Task render method already handles those different cases for you
This is where the powere of object oriented programming comes into play
*/
result += `\n|${i}> ${this.lists[i].render()}`;
```